theseus/embodied/motionmodel/misc.py

```diff
@@ -0,0 +1,160 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import List, Optional, Sequence, Tuple, Union
+
+import torch
+
+from theseus.core import CostFunction, CostWeight, Variable, as_variable
+from theseus.geometry import SE2, Point3, Vector
+
+
+class HingeLoss(CostFunction):
```

Comment:
This should be move to a more general folder (along with `th.Local` implementation).